pkg/cli/login/loginoptions.go

```diff
@@ -265,7 +265,7 @@ func (o *LoginOptions) gatherAuthInfo() error {
 	clientConfig.CertFile = o.CertFile
 	clientConfig.KeyFile = o.KeyFile
 
-	token, err := tokenrequest.RequestToken(o.Config, o.getAuthChallengeHandler())
+	token, err := tokenrequest.RequestTokenWithChallengeHandlers(o.Config, o.getAuthChallengeHandler())
```

Comment:
This PR also applies the changes that https://github.com/openshift/oc/pull/1402/ requires